docs/admin/kubeadm.md

```diff
@@ -13,188 +13,245 @@ Running `kubeadm init` bootstraps a Kubernetes cluster. This consists of the
-file manually, before running `kubeadm init`. `kubeadm` automatically
-picks those settings up and ensures other nodes are configured correctly.
-The exact format and content of the file `/etc/kubernetes/cloud-config` depends
-on the type you specified for `--cloud-provider`; see the appropriate documentation
-for your cloud provider for details.
-You must also set the `--cloud-provider` and `--cloud-config` parameters
-yourself by editing the `/etc/systemd/system/kubelet.service.d/10-kubeadm.conf`
-file appropriately.
-
-- `--external-etcd-cafile` etcd certificate authority file
-- `--external-etcd-endpoints` (multiple values are allowed)
-- `--external-etcd-certfile` etcd client certificate file
-- `--external-etcd-keyfile` etcd client key file
-
-By default, `kubeadm` deploys a single node etcd cluster on the master
-to store Kubernetes state. This means that any failure on the master node
-requires you to rebuild your cluster from scratch. Currently `kubeadm init`
-does not support automatic deployment of a highly available etcd cluster.
-If you would like to use your own etcd cluster, you can override this
-behaviour with `--external-etcd-endpoints`. `kubeadm` supports etcd client
-authentication using the `--external-etcd-cafile`, `--external-etcd-certfile`
-and `--external-etcd-keyfile` flags.
+It is usually sufficient to run `kubeadm init` without any flags, but in some
+cases you might like to override the default behaviour. Here we specify all the
+flags that can be used to customise the Kubernetes installation.
+
+- `--api-advertise-address`
+
+This is the address the API Server will advertise to other members of the
+cluster.  This is also the address used to construct the suggested `kubeadm
+join` line at the end of the init process.  If not set (or set to 0.0.0.0) then
+IP for the default interface will be used.
+
+This address is also added to the certifcate that the API Server uses.
+
+- `--apiserver-bind-port`
+
+The port that the API server will bind on.  This defaults to 6443.
+
+- `--apiserver-cert-extra-sans`
+
+Additional hostnames or IP addresses that should be added to the Subject
+Alternate Name section for the certificate that the API Server will use.  If you
+expose the API Server through a load balancer and public DNS you could specify
+this with
+
+```
+--apiserver-cert-extra-sans=kubernetes.example.com,kube.example.com,10.100.245.1
+```
+
+- `--cert-dir`
+
+The path where to save and store the certificates.  The default is
+"/etc/kubernetes/pki".
+
+- `--config`
+
+A kubeadm specific [config file](#config-file).  This can be used to specify an
+extended set of options including passing arbitrary command line flags to the
+control plane components.
+
+- `--kubernetes-version` (default 'latest') the kubernetes version to initialise
+
+The **v1.6** version of kubeadm only supports building clusters that are at
+least **v1.6.0**.  This requirement is due to kubeadm's use of RBAC. With this
```

Comment:
the Certificates API, Token Auth module, Token controllers, etc. 
There are many reasons